### ISSUE TITLE
Poorly visible header and section with error messages

### ISSUE THREAD

**Issue by nenad-vujicic** [openstreetmap/openstreetmap-website#5032]
### Problem

While working on #5023 we noticed that bar with error / warning / notice messages is often not visible, especially on small screens, so, it's not much usable in situations when page is not fully reloaded, and error occurs on some user action (in case of #5023, when user tries to Preview and body is empty).

### Description

We could improve current behavior by putting header and (or only) errors / warnings / notices bar (defined in app/views/layouts/_flash.html.erb) on the top (anchored), thus making them always visible.

What do you think is the best approach?

### Screenshots

At the moment:
![image](https://github.com/user-attachments/assets/c1c9ac49-83ee-441f-a831-e4e574ef4ccd)
Proposal:
![image](https://github.com/user-attachments/assets/64f54009-57b5-49d7-b74a-6edb07b7b66b)

**Comment by tomhughes** [comment]
No we are not going to plaster it all over the top of other UI elements.

I don't understand your first screenshot though? Do you have the page scrolled down or something? Why is the URL bar over the page content?

**Comment by nenad-vujicic** [comment]
> I don't understand your first screenshot though? Do you have the page scrolled down or something? Why is the URL bar over the page content?

Yes, I scrolled down to "simulate" the problem with current sources in master branch.

> No we are not going to plaster it all over the top of other UI elements.

OK, in that case, should I close this issue?

Also, should we scroll to the top of the page when error / warning / notice message occurs and page is not fully reloaded (so default bar cannot be used) or we should force full page reloading (in which case default bar can be used)? I believe it depends on the situation :-)

Thanks,
Nenad.

**Comment by tomhughes** [comment]
Well I think I need to understand better how to reach the problem state before I can comment.

**Comment by nenad-vujicic** [comment]
Sure, let me explain it.

I noticed this while working on PR #5023 where I updated HTML DOM tree with bootstrap alert for errors when message / diary entry / diary entry comment body is empty. Because window was too small, after pressing Preview button, bar with error message was not visible (it was too high).

On other places, alerts are placed on other places (e.g. inside of sidebar, ..), so, I'm not sure you can reproduce it with master sources, since they are already visible. But, in this case (PR #5023), it looks like the most appropriate place (this is subjective and I would like to read other's opinions) is below the header (i.e. at the beginning of the content section), especially if body is long.

Thanks,
Nenad.

**Comment by AntonKhorev** [comment]
> Because window was too small, after pressing Preview button, bar with error message was not visible (it was too high).

Because the error message you're adding in #5023 is not what I meant here https://github.com/openstreetmap/openstreetmap-website/pull/4894#issuecomment-2168418319 and now you discovered why. This is what I meant:

![image](https://github.com/user-attachments/assets/3e7687f2-09cb-485a-b194-642b54a70b51)

For the login page with autofocus on the input maybe the error message should be under the input.

**Comment by nenad-vujicic** [comment]
> Because the error message you're adding in #5023 is not what I meant here [#4894 (comment)](https://github.com/openstreetmap/openstreetmap-website/pull/4894#issuecomment-2168418319) and now you discovered why. This is what I meant:
> 

@AntonKhorev , thank you very much. I've just updated #5023 

Thanks,
Nenad.

**Comment by msudipta888** [comment]
can i solve this issue if agree

**Comment by nenad-vujicic** [comment]
I got all the answers I needed, so, yes, we can close this issue. Thanks!
